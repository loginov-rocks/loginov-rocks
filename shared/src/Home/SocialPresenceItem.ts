export interface SocialPresenceItem {
  title: string;
  url: string;
  url2?: string;
}
