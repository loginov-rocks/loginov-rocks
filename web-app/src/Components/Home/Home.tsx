import { GitHubData } from '@loginov-rocks/loginov-rocks-shared';
import * as React from 'react';

import { SocialPresence } from 'Components/SocialPresence';
import { WorksList } from 'Components/WorksList';
import { WEB_APP_S3_GITHUB_FILE_KEY } from 'Constants';
import { HomeData } from 'Data/Interfaces/HomeData';
import { GitHubData as GetGitHubData } from 'Lib/GitHubData';

const getGitHubData = new GetGitHubData({ url: `/${WEB_APP_S3_GITHUB_FILE_KEY}` });

interface Props {
  data: HomeData;
}

export const Home: React.FunctionComponent<Props> = ({ data }) => {
  const [gitHubData, setGitHubData] = React.useState<GitHubData | null>(null);

  React.useEffect(() => {
    getGitHubData.get()
      .then((_gitHubData) => {
        setGitHubData(_gitHubData);
      });
  }, []);

  return (
    <>
      <h1>Hello!</h1>
      <p>
        I am
        {' '}
        <strong>Danila</strong>
        , nice to e-meet you!
      </p>
      <>
        <h2>Social Presence</h2>
        <SocialPresence items={data.socialPresenceItems} />
      </>
      <>
        <h2>Open Source</h2>
        <>
          <h3>Projects</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.projects} />
        </>
        <>
          <h3>Libraries</h3>
          <WorksList gitHubRepos={gitHubData ? gitHubData.repos : []} works={data.openSource.libraries} />
        </>
      </>
    </>
  );
};
