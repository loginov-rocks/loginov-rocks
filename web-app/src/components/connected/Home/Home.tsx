import * as React from 'react';

import { CmsConnectedProps } from 'cms/interfaces/CmsConnectedProps';
import { CmsEntry } from 'cms/interfaces/CmsEntry';

interface Props extends CmsConnectedProps {
  sections: CmsEntry[];
  title: string;
}

export const Home: React.FC<Props> = ({ render, sections, title }) => (
  <>
    <h1>{title}</h1>
    {render(sections)}
  </>
);
