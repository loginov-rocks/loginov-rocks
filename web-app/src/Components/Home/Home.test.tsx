/* eslint-disable import/no-extraneous-dependencies */

import { shallow } from 'enzyme';
import * as React from 'react';

import * as homeData from '../../Data/Home.json';

import { Home } from './Home';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const gitHubMockData = require('@loginov-rocks/loginov-rocks-shared/src/GitHub/__fixtures__/data.json');

it('matches snapshot', () => {
  const wrapper = shallow(<Home data={homeData} gitHubData={gitHubMockData} />);

  expect(wrapper).toMatchSnapshot();
});
