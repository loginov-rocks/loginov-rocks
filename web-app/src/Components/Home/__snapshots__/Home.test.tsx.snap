// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Hello!
  </h1>
  <p>
    I am
     
    <strong>
      Danila
    </strong>
    , nice to e-meet you!
  </p>
  <h2>
    Social Presence
  </h2>
  <SocialPresence
    items={
      Array [
        Object {
          "title": "LinkedIn",
          "url": "https://www.linkedin.com/in/loginov-rocks/",
        },
        Object {
          "title": "GitHub",
          "url": "https://github.com/loginov-rocks",
        },
        Object {
          "title": "Medium",
          "url": "https://loginov-rocks.medium.com/",
        },
        Object {
          "title": "DRIVE2",
          "url": "https://www.drive2.com/r/toyota/chaser/288230376151952785/",
          "url2": "https://www.drive2.ru/r/toyota/chaser/288230376151952785/",
        },
        Object {
          "title": "Instagram",
          "url": "https://www.instagram.com/loginov_rocks/",
        },
        Object {
          "title": "Facebook",
          "url": "https://www.facebook.com/danila.rocks",
        },
        Object {
          "title": "Twitter",
          "url": "https://twitter.com/loginov_rocks",
        },
        Object {
          "title": "npm",
          "url": "https://www.npmjs.com/~loginov-rocks",
        },
        Object {
          "title": "Stack Overflow",
          "url": "https://stackoverflow.com/users/5021580/danila-loginov",
        },
        Object {
          "title": "vk",
          "url": "https://vk.com/danilaloginov",
        },
        Object {
          "title": "YouTube",
          "url": "https://www.youtube.com/c/DanilaLoginov",
        },
        Object {
          "title": "about.me",
          "url": "https://about.me/danilaloginov",
        },
        Object {
          "title": "Habr",
          "url": "https://habr.com/en/users/loginov/",
        },
      ]
    }
  />
  <h2>
    Open Source
  </h2>
  <h3>
    Projects
  </h3>
  <WorksList
    gitHubRepos={
      Array [
        Object {
          "description": "Clean but full featured AngularJS boilerplate using Gulp workflow and best practices",
          "homepageUrl": "https://www.npmjs.com/package/angular-gulp-boilerplate",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.10",
          "name": "Angular-Gulp-Boilerplate",
          "stars": 29,
          "updatedAt": 1614276648000,
          "url": "https://github.com/loginov-rocks/Angular-Gulp-Boilerplate",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.2.0",
          "name": "Anticloud",
          "stars": 0,
          "updatedAt": 1614278791000,
          "url": "https://github.com/loginov-rocks/Anticloud",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "",
          "name": "anticloud-server",
          "stars": 0,
          "updatedAt": 1614276807000,
          "url": "https://github.com/loginov-rocks/anticloud-server",
        },
        Object {
          "description": "ES6 class for serial communication with your own Bluetooth Low Energy (Smart) devices",
          "homepageUrl": "https://loginov-rocks.github.io/Web-Bluetooth-Terminal/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.4.2",
          "name": "bluetooth-terminal",
          "stars": 31,
          "updatedAt": 1610847914000,
          "url": "https://github.com/loginov-rocks/bluetooth-terminal",
        },
        Object {
          "description": "Canvas heightmap generator",
          "homepageUrl": "https://www.npmjs.com/package/canvas-heightmap",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.7",
          "name": "canvas-heightmap",
          "stars": 1,
          "updatedAt": 1609888189000,
          "url": "https://github.com/loginov-rocks/canvas-heightmap",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.3.0",
          "name": "Carbook",
          "stars": 0,
          "updatedAt": 1614278890000,
          "url": "https://github.com/loginov-rocks/Carbook",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "Dart",
          "latestTag": "",
          "name": "carquery",
          "stars": 0,
          "updatedAt": 1614274482000,
          "url": "https://github.com/loginov-rocks/carquery",
        },
        Object {
          "description": "Extract comments from glob of files, annotated or not",
          "homepageUrl": "https://www.npmjs.com/package/comments-extractor",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.10",
          "name": "comments-extractor",
          "stars": 0,
          "updatedAt": 1609802038000,
          "url": "https://github.com/loginov-rocks/comments-extractor",
        },
        Object {
          "description": "",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "Connected-Espresso-Machine",
          "stars": 0,
          "updatedAt": 1614276068000,
          "url": "https://github.com/loginov-rocks/Connected-Espresso-Machine",
        },
        Object {
          "description": "Simple app for communication with devices via Bluetooth based on Apache Cordova (PhoneGap)",
          "homepageUrl": "",
          "isArchived": true,
          "language": "JavaScript",
          "latestTag": "v1.1.1",
          "name": "Cordova-Bluetooth-Terminal",
          "stars": 10,
          "updatedAt": 1614279642000,
          "url": "https://github.com/loginov-rocks/Cordova-Bluetooth-Terminal",
        },
        Object {
          "description": "ðŸ¦‹ An OpenApi validator for ExpressJS that automatically validates API requests using an OpenAPI 3.x specification,",
          "homepageUrl": "",
          "isArchived": true,
          "language": "TypeScript",
          "latestTag": "v1.6.2",
          "name": "express-openapi-validator",
          "stars": 0,
          "updatedAt": 1614276944000,
          "url": "https://github.com/loginov-rocks/express-openapi-validator",
        },
        Object {
          "description": "Firebase Authenticator",
          "homepageUrl": "https://loginov-rocks.github.io/Firebase-Authenticator/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.2.2",
          "name": "Firebase-Authenticator",
          "stars": 0,
          "updatedAt": 1614276466000,
          "url": "https://github.com/loginov-rocks/Firebase-Authenticator",
        },
        Object {
          "description": "Gulp plugin to bundle JSON localization files",
          "homepageUrl": "https://www.npmjs.com/package/gulp-locales-bundler",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.7",
          "name": "gulp-locales-bundler",
          "stars": 1,
          "updatedAt": 1609889891000,
          "url": "https://github.com/loginov-rocks/gulp-locales-bundler",
        },
        Object {
          "description": "Documentation generator for Gulp tasks",
          "homepageUrl": "https://www.npmjs.com/package/gulpdoc",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.1.7",
          "name": "gulpdoc",
          "stars": 0,
          "updatedAt": 1609802952000,
          "url": "https://github.com/loginov-rocks/gulpdoc",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "",
          "name": "Igla",
          "stars": 0,
          "updatedAt": 1614274228000,
          "url": "https://github.com/loginov-rocks/Igla",
        },
        Object {
          "description": "",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "IoT-Firebase-Authentication",
          "stars": 0,
          "updatedAt": 1614274532000,
          "url": "https://github.com/loginov-rocks/IoT-Firebase-Authentication",
        },
        Object {
          "description": "Script to merge deep hierarchy of JSON files into a flat set of objects (files) containing deep objects inside",
          "homepageUrl": "https://www.npmjs.com/package/locales-bundler",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.3",
          "name": "locales-bundler",
          "stars": 0,
          "updatedAt": 1614332403000,
          "url": "https://github.com/loginov-rocks/locales-bundler",
        },
        Object {
          "description": "Monorepo for my personal website",
          "homepageUrl": "https://loginov.rocks",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v2.3.1",
          "name": "loginov-rocks",
          "stars": 0,
          "updatedAt": 1614463481000,
          "url": "https://github.com/loginov-rocks/loginov-rocks",
        },
        Object {
          "description": "MeArm Controller",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "v0.1.1",
          "name": "MeArm-Controller",
          "stars": 2,
          "updatedAt": 1614274371000,
          "url": "https://github.com/loginov-rocks/MeArm-Controller",
        },
        Object {
          "description": "MQTT Home Monitor",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "MQTT-Home-Monitor",
          "stars": 0,
          "updatedAt": 1614276718000,
          "url": "https://github.com/loginov-rocks/MQTT-Home-Monitor",
        },
        Object {
          "description": "Monorepo for an application built completely with JavaScript, based on React and Nest frameworks and combining serverless and containerized approach on AWS",
          "homepageUrl": "https://rs.school/nodejs-aws/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "Node-AWS-App",
          "stars": 0,
          "updatedAt": 1614330404000,
          "url": "https://github.com/loginov-rocks/Node-AWS-App",
        },
        Object {
          "description": "Quick start with NodeMCU v3 (ESP8266), Arduino ecosystem, and PlatformIO IDE",
          "homepageUrl": "https://loginov-rocks.medium.com/quick-start-with-nodemcu-v3-esp8266-arduino-ecosystem-and-platformio-ide-b8415bf9a038",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "NodeMCU-Arduino-PlatformIO",
          "stars": 2,
          "updatedAt": 1614276030000,
          "url": "https://github.com/loginov-rocks/NodeMCU-Arduino-PlatformIO",
        },
        Object {
          "description": "Ooyala Player Component",
          "homepageUrl": "https://www.npmjs.com/package/ooyala-player-component",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.1.2",
          "name": "ooyala-player-component",
          "stars": 0,
          "updatedAt": 1609892621000,
          "url": "https://github.com/loginov-rocks/ooyala-player-component",
        },
        Object {
          "description": "Portfolio App",
          "homepageUrl": "https://loginov-rocks.github.io/Portfolio/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v1.1.2",
          "name": "Portfolio",
          "stars": 0,
          "updatedAt": 1610903025000,
          "url": "https://github.com/loginov-rocks/Portfolio",
        },
        Object {
          "description": "Firebase Services for Portfolio App",
          "homepageUrl": "https://loginov-rocks.github.io/Portfolio/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.2.2",
          "name": "Portfolio-Firebase",
          "stars": 0,
          "updatedAt": 1614276530000,
          "url": "https://github.com/loginov-rocks/Portfolio-Firebase",
        },
        Object {
          "description": "React Native Redux Form fields",
          "homepageUrl": "https://www.npmjs.com/package/react-native-redux-form",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.1.3",
          "name": "react-native-redux-form",
          "stars": 0,
          "updatedAt": 1609890994000,
          "url": "https://github.com/loginov-rocks/react-native-redux-form",
        },
        Object {
          "description": "Redux Repository",
          "homepageUrl": "https://www.npmjs.com/package/redux-repository",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.6.2",
          "name": "redux-repository",
          "stars": 0,
          "updatedAt": 1609893885000,
          "url": "https://github.com/loginov-rocks/redux-repository",
        },
        Object {
          "description": "Three.js Sky object from official repository example",
          "homepageUrl": "https://www.npmjs.com/package/three-sky",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.1.3",
          "name": "three-sky",
          "stars": 2,
          "updatedAt": 1609888944000,
          "url": "https://github.com/loginov-rocks/three-sky",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "todogle",
          "stars": 0,
          "updatedAt": 1614276613000,
          "url": "https://github.com/loginov-rocks/todogle",
        },
        Object {
          "description": "Arduino library for the fastest and simplest communication with u-blox GPS modules",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "v1.4.0",
          "name": "UbxGps",
          "stars": 83,
          "updatedAt": 1614274887000,
          "url": "https://github.com/loginov-rocks/UbxGps",
        },
        Object {
          "description": "Progressive Web Application for serial communication with your own Bluetooth Low Energy (Smart) devices",
          "homepageUrl": "https://loginov-rocks.github.io/Web-Bluetooth-Terminal/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.3.3",
          "name": "Web-Bluetooth-Terminal",
          "stars": 129,
          "updatedAt": 1614276407000,
          "url": "https://github.com/loginov-rocks/Web-Bluetooth-Terminal",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "ZenMoney-Portfolio-Calculator",
          "stars": 1,
          "updatedAt": 1614274813000,
          "url": "https://github.com/loginov-rocks/ZenMoney-Portfolio-Calculator",
        },
      ]
    }
    works={
      Array [
        "Web-Bluetooth-Terminal",
        "Portfolio",
        "Firebase-Authenticator",
        "Angular-Gulp-Boilerplate",
        "Connected-Espresso-Machine",
        "NodeMCU-Arduino-PlatformIO",
        "loginov-rocks",
        "Node-AWS-App",
        "Cordova-Bluetooth-Terminal",
        "AWS-IoT-GSM-Modem",
        "Build-Lambda-in-Docker",
      ]
    }
  />
  <h3>
    Libraries
  </h3>
  <WorksList
    gitHubRepos={
      Array [
        Object {
          "description": "Clean but full featured AngularJS boilerplate using Gulp workflow and best practices",
          "homepageUrl": "https://www.npmjs.com/package/angular-gulp-boilerplate",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.10",
          "name": "Angular-Gulp-Boilerplate",
          "stars": 29,
          "updatedAt": 1614276648000,
          "url": "https://github.com/loginov-rocks/Angular-Gulp-Boilerplate",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.2.0",
          "name": "Anticloud",
          "stars": 0,
          "updatedAt": 1614278791000,
          "url": "https://github.com/loginov-rocks/Anticloud",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "",
          "name": "anticloud-server",
          "stars": 0,
          "updatedAt": 1614276807000,
          "url": "https://github.com/loginov-rocks/anticloud-server",
        },
        Object {
          "description": "ES6 class for serial communication with your own Bluetooth Low Energy (Smart) devices",
          "homepageUrl": "https://loginov-rocks.github.io/Web-Bluetooth-Terminal/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.4.2",
          "name": "bluetooth-terminal",
          "stars": 31,
          "updatedAt": 1610847914000,
          "url": "https://github.com/loginov-rocks/bluetooth-terminal",
        },
        Object {
          "description": "Canvas heightmap generator",
          "homepageUrl": "https://www.npmjs.com/package/canvas-heightmap",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.7",
          "name": "canvas-heightmap",
          "stars": 1,
          "updatedAt": 1609888189000,
          "url": "https://github.com/loginov-rocks/canvas-heightmap",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.3.0",
          "name": "Carbook",
          "stars": 0,
          "updatedAt": 1614278890000,
          "url": "https://github.com/loginov-rocks/Carbook",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "Dart",
          "latestTag": "",
          "name": "carquery",
          "stars": 0,
          "updatedAt": 1614274482000,
          "url": "https://github.com/loginov-rocks/carquery",
        },
        Object {
          "description": "Extract comments from glob of files, annotated or not",
          "homepageUrl": "https://www.npmjs.com/package/comments-extractor",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.10",
          "name": "comments-extractor",
          "stars": 0,
          "updatedAt": 1609802038000,
          "url": "https://github.com/loginov-rocks/comments-extractor",
        },
        Object {
          "description": "",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "Connected-Espresso-Machine",
          "stars": 0,
          "updatedAt": 1614276068000,
          "url": "https://github.com/loginov-rocks/Connected-Espresso-Machine",
        },
        Object {
          "description": "Simple app for communication with devices via Bluetooth based on Apache Cordova (PhoneGap)",
          "homepageUrl": "",
          "isArchived": true,
          "language": "JavaScript",
          "latestTag": "v1.1.1",
          "name": "Cordova-Bluetooth-Terminal",
          "stars": 10,
          "updatedAt": 1614279642000,
          "url": "https://github.com/loginov-rocks/Cordova-Bluetooth-Terminal",
        },
        Object {
          "description": "ðŸ¦‹ An OpenApi validator for ExpressJS that automatically validates API requests using an OpenAPI 3.x specification,",
          "homepageUrl": "",
          "isArchived": true,
          "language": "TypeScript",
          "latestTag": "v1.6.2",
          "name": "express-openapi-validator",
          "stars": 0,
          "updatedAt": 1614276944000,
          "url": "https://github.com/loginov-rocks/express-openapi-validator",
        },
        Object {
          "description": "Firebase Authenticator",
          "homepageUrl": "https://loginov-rocks.github.io/Firebase-Authenticator/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.2.2",
          "name": "Firebase-Authenticator",
          "stars": 0,
          "updatedAt": 1614276466000,
          "url": "https://github.com/loginov-rocks/Firebase-Authenticator",
        },
        Object {
          "description": "Gulp plugin to bundle JSON localization files",
          "homepageUrl": "https://www.npmjs.com/package/gulp-locales-bundler",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.7",
          "name": "gulp-locales-bundler",
          "stars": 1,
          "updatedAt": 1609889891000,
          "url": "https://github.com/loginov-rocks/gulp-locales-bundler",
        },
        Object {
          "description": "Documentation generator for Gulp tasks",
          "homepageUrl": "https://www.npmjs.com/package/gulpdoc",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.1.7",
          "name": "gulpdoc",
          "stars": 0,
          "updatedAt": 1609802952000,
          "url": "https://github.com/loginov-rocks/gulpdoc",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "",
          "name": "Igla",
          "stars": 0,
          "updatedAt": 1614274228000,
          "url": "https://github.com/loginov-rocks/Igla",
        },
        Object {
          "description": "",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "IoT-Firebase-Authentication",
          "stars": 0,
          "updatedAt": 1614274532000,
          "url": "https://github.com/loginov-rocks/IoT-Firebase-Authentication",
        },
        Object {
          "description": "Script to merge deep hierarchy of JSON files into a flat set of objects (files) containing deep objects inside",
          "homepageUrl": "https://www.npmjs.com/package/locales-bundler",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.0.3",
          "name": "locales-bundler",
          "stars": 0,
          "updatedAt": 1614332403000,
          "url": "https://github.com/loginov-rocks/locales-bundler",
        },
        Object {
          "description": "Monorepo for my personal website",
          "homepageUrl": "https://loginov.rocks",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v2.3.1",
          "name": "loginov-rocks",
          "stars": 0,
          "updatedAt": 1614463481000,
          "url": "https://github.com/loginov-rocks/loginov-rocks",
        },
        Object {
          "description": "MeArm Controller",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "v0.1.1",
          "name": "MeArm-Controller",
          "stars": 2,
          "updatedAt": 1614274371000,
          "url": "https://github.com/loginov-rocks/MeArm-Controller",
        },
        Object {
          "description": "MQTT Home Monitor",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "MQTT-Home-Monitor",
          "stars": 0,
          "updatedAt": 1614276718000,
          "url": "https://github.com/loginov-rocks/MQTT-Home-Monitor",
        },
        Object {
          "description": "Monorepo for an application built completely with JavaScript, based on React and Nest frameworks and combining serverless and containerized approach on AWS",
          "homepageUrl": "https://rs.school/nodejs-aws/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "Node-AWS-App",
          "stars": 0,
          "updatedAt": 1614330404000,
          "url": "https://github.com/loginov-rocks/Node-AWS-App",
        },
        Object {
          "description": "Quick start with NodeMCU v3 (ESP8266), Arduino ecosystem, and PlatformIO IDE",
          "homepageUrl": "https://loginov-rocks.medium.com/quick-start-with-nodemcu-v3-esp8266-arduino-ecosystem-and-platformio-ide-b8415bf9a038",
          "isArchived": false,
          "language": "C++",
          "latestTag": "",
          "name": "NodeMCU-Arduino-PlatformIO",
          "stars": 2,
          "updatedAt": 1614276030000,
          "url": "https://github.com/loginov-rocks/NodeMCU-Arduino-PlatformIO",
        },
        Object {
          "description": "Ooyala Player Component",
          "homepageUrl": "https://www.npmjs.com/package/ooyala-player-component",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.1.2",
          "name": "ooyala-player-component",
          "stars": 0,
          "updatedAt": 1609892621000,
          "url": "https://github.com/loginov-rocks/ooyala-player-component",
        },
        Object {
          "description": "Portfolio App",
          "homepageUrl": "https://loginov-rocks.github.io/Portfolio/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v1.1.2",
          "name": "Portfolio",
          "stars": 0,
          "updatedAt": 1610903025000,
          "url": "https://github.com/loginov-rocks/Portfolio",
        },
        Object {
          "description": "Firebase Services for Portfolio App",
          "homepageUrl": "https://loginov-rocks.github.io/Portfolio/",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.2.2",
          "name": "Portfolio-Firebase",
          "stars": 0,
          "updatedAt": 1614276530000,
          "url": "https://github.com/loginov-rocks/Portfolio-Firebase",
        },
        Object {
          "description": "React Native Redux Form fields",
          "homepageUrl": "https://www.npmjs.com/package/react-native-redux-form",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v0.1.3",
          "name": "react-native-redux-form",
          "stars": 0,
          "updatedAt": 1609890994000,
          "url": "https://github.com/loginov-rocks/react-native-redux-form",
        },
        Object {
          "description": "Redux Repository",
          "homepageUrl": "https://www.npmjs.com/package/redux-repository",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "v0.6.2",
          "name": "redux-repository",
          "stars": 0,
          "updatedAt": 1609893885000,
          "url": "https://github.com/loginov-rocks/redux-repository",
        },
        Object {
          "description": "Three.js Sky object from official repository example",
          "homepageUrl": "https://www.npmjs.com/package/three-sky",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.1.3",
          "name": "three-sky",
          "stars": 2,
          "updatedAt": 1609888944000,
          "url": "https://github.com/loginov-rocks/three-sky",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "todogle",
          "stars": 0,
          "updatedAt": 1614276613000,
          "url": "https://github.com/loginov-rocks/todogle",
        },
        Object {
          "description": "Arduino library for the fastest and simplest communication with u-blox GPS modules",
          "homepageUrl": "",
          "isArchived": false,
          "language": "C++",
          "latestTag": "v1.4.0",
          "name": "UbxGps",
          "stars": 83,
          "updatedAt": 1614274887000,
          "url": "https://github.com/loginov-rocks/UbxGps",
        },
        Object {
          "description": "Progressive Web Application for serial communication with your own Bluetooth Low Energy (Smart) devices",
          "homepageUrl": "https://loginov-rocks.github.io/Web-Bluetooth-Terminal/",
          "isArchived": false,
          "language": "JavaScript",
          "latestTag": "v1.3.3",
          "name": "Web-Bluetooth-Terminal",
          "stars": 129,
          "updatedAt": 1614276407000,
          "url": "https://github.com/loginov-rocks/Web-Bluetooth-Terminal",
        },
        Object {
          "description": "Training Repository",
          "homepageUrl": "",
          "isArchived": false,
          "language": "TypeScript",
          "latestTag": "",
          "name": "ZenMoney-Portfolio-Calculator",
          "stars": 1,
          "updatedAt": 1614274813000,
          "url": "https://github.com/loginov-rocks/ZenMoney-Portfolio-Calculator",
        },
      ]
    }
    works={
      Array [
        "UbxGps",
        "bluetooth-terminal",
        "canvas-heightmap",
        "comments-extractor",
        "gulpdoc",
        "gulp-locales-bundler",
        "locales-bundler",
        "react-native-redux-form",
        "redux-repository",
        "three-sky",
        "ooyala-player-component",
      ]
    }
  />
</Fragment>
`;
